language/apply.md

```diff
@@ -0,0 +1,34 @@
+Apply Expression
+===
+
+Apply is an expression intended to capture a manifest block, compile it to a catalog, and apply it on remote nodes.
+
+Grammar
+---
+
+    ApplyExpression
+      : 'apply' arguments = ArgumentList '{' statements? '}'
+      ;
+
+    statements
+       # All statements in the Puppet Programming Language (not shown) except NodeDefinition,
+       # ClassDefinition, ResourceTypeDefinition, FuntionDefinition, and exported resource collectors
+       : ...
+       | primary_expression
+
+    primary_expression
+      : ... # all expressions that are primary expression in the Puppet Language (not shown)
+      | epp_render_expression
+
+See [Function Calls](expressions.md#function-calls) for `ArgumentList` grammar definitions.
+
+An ApplyExpression takes an argument list requiring at least zero or more arguments. The intention is that the 1st
```

Comment:
Formatting of ApplyExpression, should probably be `ApplyExpression`

Reply:
Saying "requiring at least zero or more" does not really make sense since that means it does not actually require any...